guvnor-m2repo-editor/guvnor-m2repo-editor-client/src/main/java/org/guvnor/m2repo/client/upload/UploadFormViewImpl.java

```diff
@@ -93,87 +95,36 @@ public WellForm doUploadForm() {
-                if ( fileName == null || "".equals( fileName ) ) {
-                    BusyPopup.close();
-                    Window.alert( M2RepoEditorConstants.INSTANCE.SelectFileUpload() );
-                    event.cancel();
-                } else if ( !( isValid( fileName ) ) ) {
-                    BusyPopup.close();
-                    Window.alert( M2RepoEditorConstants.INSTANCE.UnsupportedFileType() );
-                    event.cancel();
-                }
+            public void execute() {
+                form.submit();
             }
+        });
 
-        } );
-
-        form.addSubmitCompleteHandler( new Form.SubmitCompleteHandler() {
-            public void onSubmitComplete( final Form.SubmitCompleteEvent event ) {
-                if ( UPLOAD_OK.equalsIgnoreCase( event.getResults() ) ) {
-                    BusyPopup.close();
-                    Window.alert( M2RepoEditorConstants.INSTANCE.UploadedSuccessfully() );
-                    hiddenFieldsPanel.setVisible( false );
-                    hiddenArtifactIdField.setText( null );
-                    hiddenGroupIdField.setText( null );
-                    hiddenVersionIdField.setText( null );
-
-                    searchEvent.fire( new M2RepoSearchEvent() );
-
-                    up.getElement().setPropertyString( "value",
-                                                       "" );
-                    hide();
-
-                } else if ( UPLOAD_MISSING_POM.equalsIgnoreCase( event.getResults() ) ) {
-                    BusyPopup.close();
-                    Window.alert( M2RepoEditorConstants.INSTANCE.InvalidJarNotPom() );
-                    hiddenFieldsPanel.setVisible( true );
-
-                } else if ( UPLOAD_UNABLE_TO_PARSE_POM.equalsIgnoreCase( event.getResults() ) ) {
-                    BusyPopup.close();
-                    Window.alert( M2RepoEditorConstants.INSTANCE.InvalidPom() );
-                    hide();
-
-                } else {
-                    BusyPopup.close();
-                    ErrorPopup.showMessage( M2RepoEditorConstants.INSTANCE.UploadFailed() + event.getResults() );
-
-                    hiddenFieldsPanel.setVisible( false );
-                    hiddenArtifactIdField.setText( null );
-                    hiddenGroupIdField.setText( null );
-                    hiddenVersionIdField.setText( null );
-                    hide();
-                }
+        uploader.setName(HTMLFileManagerFields.UPLOAD_FIELD_NAME_ATTACH);
+
+        form.addSubmitCompleteHandler(new Form.SubmitCompleteHandler() {
+            public void onSubmitComplete(final Form.SubmitCompleteEvent event) {
+                presenter.handleSubmitComplete(event);
             }
-        } );
+        });
 
         HorizontalPanel fields = new HorizontalPanel();
-        fields.add( up );
+        fields.add(uploader);
 
         hiddenGroupIdField.setName( HTMLFileManagerFields.GROUP_ID );
-        hiddenGroupIdField.setText( null );
```

Comment:
It was duplication (about 3 times in one file) so I am moved it to separate method.

hiddenGroupIdField.setText( null );
iddenArtifactIdField.setText( null );
hiddenVersionIdField.setText( null );
hiddenFieldsPanel.setVisible( false );